packages/flutter_tools/templates/plugin/android-java.tmpl/src/main/java/androidIdentifier/pluginClass.java.tmpl

```diff
@@ -21,6 +22,20 @@ public class {{pluginClass}} implements FlutterPlugin, MethodCallHandler {
     channel.setMethodCallHandler(new {{pluginClass}}());
   }
 
+  // This function is optional and equivalent to onAttachedToEngine. It supports the old
+  // pre-Flutter-1.12 Android projects. Plugin developers are encouraged to continue supporting
+  // plugin registration via this function while apps migrate to use the new Android APIs
+  // post-flutter-1.12.
+  //
+  // Developers are encouraged to share logic between onAttachedToEngine and registerWith to keep
```

Comment:
Should we link to a canonical example or a guide?

Reply:
+1, but we don't have a guide here that's good enough to link to yet. The Wiki is for Flutter developers ourselves and not Flutter users. The guide on our wiki is great for our purposes but we'd need to move it to the website and probably clean it up some first. I think writing a comprehensive tutorial would take some time.

Reply:
re: examples

If this is about how to share logic, I think it's generic enough that it doesn't need to be explained in a code comment. And the template code already does it. 

If it's about how to migrate an old plugin to a new plugin, we should document it in https://github.com/flutter/flutter/issues/44224 and https://github.com/flutter/flutter/issues/44225. Though I don't think we need to link those in this file. The users just made a new plugin here. They're not trying to migrate a plugin at this moment. 

@mklim, comment may or may not be orthogonal. https://github.com/flutter/flutter/issues/44224 we should create a proper flutter.dev guide for the migration. Though we don't need to do it this week.